src/platform/packages/shared/kbn-esql-ast/src/parser/cst_to_ast_converter.ts

```diff
@@ -1054,8 +1054,8 @@ export class CstToAstConverter {
     const onOption = this.toOption('on', joinCondition);
     const joinPredicates: ast.ESQLAstItem[] = onOption.args;
 
-    for (const joinPredicateCtx of joinCondition.joinPredicate_list()) {
-      const expression = this.visitValueExpression(joinPredicateCtx.valueExpression());
+    for (const joinPredicateCtx of joinCondition.booleanExpression_list()) {
```

Comment:
I think with this change I did we dont need anything else to make JOIN work with the changes (meaning that it works as it used to and we dont incorporate the expressions yet) but let's check it first to be sure